File: MapA7/MapA7/Repository/Repository.cs
using MapA7.Model;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MapA7.Repository
{
    public class Repositoryx:IRepository
    {
        private List<ProgramState> programs;
        private String filepath=null;

        public Repositoryx()
        {
            programs = new List<ProgramState>();
        }

        public void SetFilepath(String filepath)
        {
            this.filepath = filepath;
        }

        public void SaveLog(ProgramState ps)
        {
            FileStream fs = new FileStream(filepath, FileMode.Append, FileAccess.Write);
            StreamWriter log = new StreamWriter(fs);
            log.WriteLine("===========================");
            log.WriteLine(ps.GetExecutionStack().Tostring());
            log.WriteLine(ps.GetSymbolTable().Tostring());
            log.WriteLine(ps.GetFileTalbe().Tostring());
            log.WriteLine(ps.GetOutput().Tostring());
            log.WriteLine("===========================");
            log.Close();
        }

        public void Add(ProgramState ps)
        {
            programs.Add(ps);
        }


        public ProgramState GetProgramState(int index)
        {
            return programs[index];
        }
    }
}

File: MapGUI3/src/Repository/Repository.java
package Repository;

import Exceptions.CustomException;
import Model.ProgramState;

import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class Repository implements IRepository {
    private String filepath = null;
    private List<ProgramState> program;

    public Repository() {
        program = new ArrayList<>();
    }

    public void setFilepath(String filepath) {
        this.filepath = filepath;
    }

    public void logPrgState(ProgramState ps) {
        try {
            PrintWriter logFile = new PrintWriter(new BufferedWriter(new FileWriter(filepath, true)));
            logFile.println("===============================");
            logFile.println("ID: "+ps.getID());
            logFile.println(ps.getExecutionStack().toString());
            logFile.println(ps.getSymTable().toString());
            logFile.println(ps.getFileTable().toString());
            logFile.println(ps.getHeap().toString());
            logFile.println(ps.getOutput().toString());
            logFile.println("===============================");
            logFile.close();
        }
        catch (IOException exc) {
            throw new CustomException("Error during log save.");
        }
    }

    public List<ProgramState> getProgramList(){return program;}
    public void setProgramList(List<ProgramState> lst) {program=lst;}

    public void add(ProgramState ps) {
        program.add(ps);
    }

    public ProgramState getProgramState(int index) {
            return program.get(index);
    }
}
